src/Core/Content/Product/Cms/ProductSlider/ProductStreamProcessor.php

```diff
@@ -154,7 +161,10 @@ private function collectFinalProductIds(ProductCollection $streamResult): array
             $finalProductIds[] = $productId ?? $product->getId();
         }
 
-        return array_unique($finalProductIds);
+        /** @var list<string> $finalIds */
+        $finalIds = array_unique($finalProductIds);
```

Comment:
```suggestion
        $finalIds = array_values(array_unique($finalProductIds));
```
then you do not need the var annotation

Reply:
ah 💡 thanks telgi